File: check_parquet.R
library(arrow)
library(dplyr)

# Check what's in the parquet file
uni45k <- read_parquet("School-Unique-v2.parquet")

cat("=== School-Unique-v2.parquet ===\n")
cat("Column names containing 'Lat' or 'Long':\n")
lat_long_cols <- names(uni45k)[grepl("Lat|Long", names(uni45k), ignore.case = TRUE)]
print(lat_long_cols)

if (length(lat_long_cols) > 0) {
  cat("\nFirst 10 rows with coordinates:\n")
  coord_data <- uni45k[, c("SchoolID", lat_long_cols)]
  print(head(coord_data[complete.cases(coord_data), ], 10))
  
  cat("\nNumber of non-NA coordinate rows:", sum(complete.cases(coord_data)), "\n")
}

File: check_parquet.py
import pyarrow.parquet as pq

# Read parquet file
table = pq.read_table('School-Unique-v2.parquet')
df = table.to_pandas()

print("=== School-Unique-v2.parquet ===")
print("\nColumns containing 'Lat' or 'Long':")
lat_long_cols = [col for col in df.columns if 'lat' in col.lower() or 'long' in col.lower()]
print(lat_long_cols)

if lat_long_cols:
    print(f"\nFirst 10 rows with coordinates:")
    print(df[['SchoolID'] + lat_long_cols].head(10))
    print(f"\nNon-null counts:")
    for col in lat_long_cols:
        print(f"{col}: {df[col].notna().sum()}")
